### PR TITLE
Add income categories

### PR DESCRIPTION
**Pull request by EugeneJ9**

### CODE REVIEW

File: client/src/components/Categories.jsx [hunk 1]
```diff
@@ -1,8 +1,111 @@
...
+    IncomeCategoriesDataService.create(data)
+      .then((response) => {
+        this.setState({
+          id: response.data.id,
+          name: response.data.name,
+          description: response.data.description,
+
+          submitted: true,
+        });
+        console.log(response.data);
```
Comment (tarzak) [inline client/src/components/Categories.jsx]: i would remove this console.log, its bad habit leave console.logs in source code

File: client/src/components/Categories.jsx [hunk 2]
```diff
@@ -1,8 +1,111 @@
...
 export default class Categories extends Component {
+  constructor(props) {
+    super(props);
+    this.onChangeName = this.onChangeName.bind(this);
+    this.onChangeDescription = this.onChangeDescription.bind(this);
+    this.saveCategory = this.saveCategory.bind(this);
+    this.newCategory = this.newCategory.bind(this);
+
+    this.state = {
+      id: null,
```
Comment (tarzak) [inline client/src/components/Categories.jsx]: we don't need this id in state, so you can remove it

File: client/src/components/Categories.jsx [hunk 3]
```diff
@@ -1,8 +1,111 @@
...
+  saveCategory() {
+    const data = {
+      name: this.state.name,
+      description: this.state.description,
+    };
+
+    IncomeCategoriesDataService.create(data)
+      .then((response) => {
+        this.setState({
+          id: response.data.id,
```
Comment (tarzak) [inline client/src/components/Categories.jsx]: this is redundant too

**Comment by EugeneJ9** [comment]
Done! Thanks!

**Pull request merged**